### DISCUSSION TITLE
How are initoverlayfs expected to be discovered?

### DISCUSSION THREAD

**Discussion by cgwalters** [containers/initoverlayfs#69] [Q&A]
One thing I don't understand from reading the docs/code is: From the initramfs, how is the initoverlayfs expected to be discovered?

Do we expect the initramfs to have the partition of `/boot` (if present) and the path to the initoverlayfs?

There's a config file I see but...oh wait, I see now it's in

```
    if (asprintf(&fs, "/boot/initoverlayfs-%s.img", buf.release) < 0)
      return -1;
```

or so.  

This type of thing would be good to add to the docs for how it works.

**Comment by ericcurtin** [comment]
initoverlayfs has it's own config file (it's in BLS format, because that's simple to parse), this is included in initrd, initoverlayfs and rootfs, here is an example one:

```
bootfs UUID=156f0420-627b-4151-ae6f-fda298097515
bootfstype ext4
initoverlayfs_builder dracut -M -o "initoverlayfs fcoe"
initrd_builder dracut -M -m "kernel-modules udev-rules initoverlayfs systemd base" -o "bash systemd-initrd i18n kernel-modules-extra rootfs-block dracut-systemd usrmount fs-lib microcode_ctl-fw_dir_override shutdown nss-softokn"
```

bootfs takes arguments in a similar format to root= on the cmdline.

We didn't use cmdline because I've ran into more that a couple of cmdline issues and just putting data in a file in the signed initrd is basically the same thing, but you get your own namespace (the file).

We will be adding more docs...

**Reply by ericcurtin** [reply]
And the initoverlayfs file has the same naming format as initramfs typically does, of course if needs be, anything can be made configurable in future:

```
/boot/initoverlayfs-5.14.0-420.373.el9iv.x86_64.img
/boot/initramfs-5.14.0-420.373.el9iv.x86_64.img
```

**Reply by ericcurtin** [reply]
We mount /boot in initramfs, setup the loopback initoverlayfs mount (with dm-verity optional), switchroot to that... The rest of the boot behaves as is...
